app/scripts/services/templates.js

```diff
@@ -0,0 +1,53 @@
+'use strict';
+
+angular.module("openshiftConsole")
+  .service("CachedTemplateService", function(){
+
+    var template = null;
+    return {
+      setTemplate: function(temp) {
+        template = temp;
+      },
+      getTemplate: function() {
+        return template;
+      },
+      clearTemplate: function() {
+        template = null;
+      }
+    };
+
+  }).service("ProcessedParametersService", function(){
+
+    var params = [];
+    var paramsWithGeneratedValues = [];
+    return {
+      setParams: function(paramaters) {
+        _.each(paramaters, function(param) {
+          params.push({
+            name: param.name,
+            value: param.value
```

Comment:
i think id rather see one pair of set/get methods and just have a 'wasGenerated' or 'generated' added to this map